rflx/expression.py

```diff
@@ -40,26 +40,33 @@ class ProofResult(Enum):
 
 
 class Proof:
+    LOGIC = "QF_NIA"
+
     def __init__(self, expr: "Expr", facts: Optional[Sequence["Expr"]] = None):
         self.__expr = expr
         self.__facts = facts or []
         self.__result = ProofResult.UNSAT
+        self.__unknown_reason = "unknown"
 
-        solver = z3.Solver()
+        solver = z3.SolverFor(self.LOGIC)
         solver.add(self.__expr.z3expr())
         for f in self.__facts:
             solver.add(f.z3expr())
 
         self.__result = ProofResult(solver.check())
+        if self.__result == ProofResult.UNKNOWN:
+            self.__unknown_reason = solver.reason_unknown()
 
     @property
     def result(self) -> ProofResult:
         return self.__result
 
     @property
     def error(self) -> List[Tuple[str, Optional[Location]]]:
+        if self.__result == ProofResult.UNKNOWN:
+            return [(self.__unknown_reason, None)]
```

Comment:
Would it make sense to return the location of `self.__expr` to have at least any location in this case?

Reply:
No, I tried it with the DHCP example which made me aware of the issue and it gave me no location either. We use `expr.proof(facts)` most of the time where `expr` is constructed (or just `TRUE`) and thus has no location.

Reply:
I see.